internal/civisibility/utils/net/settings_api.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package net
+
+import (
+	"fmt"
+)
+
+const (
+	settingsRequestType string = "ci_app_test_service_libraries_settings"
+	settingsURLPath     string = "api/v2/libraries/tests/services/setting"
+)
+
+type (
+	settingsRequest struct {
+		Data settingsRequestHeader `json:"data,omitempty"`
+	}
+
+	settingsRequestHeader struct {
+		ID         string              `json:"id,omitempty"`
+		Type       string              `json:"type,omitempty"`
+		Attributes SettingsRequestData `json:"attributes,omitempty"`
+	}
+
+	SettingsRequestData struct {
+		Service        string             `json:"service,omitempty"`
+		Env            string             `json:"env,omitempty"`
+		RepositoryURL  string             `json:"repository_url,omitempty"`
+		Branch         string             `json:"branch,omitempty"`
+		Sha            string             `json:"sha,omitempty"`
+		Configurations testConfigurations `json:"configurations,omitempty"`
+	}
+
+	settingsResponse struct {
+		Data struct {
+			ID         string               `json:"id,omitempty"`
+			Type       string               `json:"type,omitempty"`
+			Attributes SettingsResponseData `json:"attributes,omitempty"`
+		} `json:"data,omitempty"`
+	}
+
+	SettingsResponseData struct {
+		CodeCoverage        bool `json:"code_coverage,omitempty"`
+		EarlyFlakeDetection struct {
+			Enabled         bool `json:"enabled,omitempty"`
+			SlowTestRetries struct {
+				One0S   int `json:"10s,omitempty"`
```

Comment:
`TenS` (and `ThirtyS` below?) 😅

Reply:
ups, done in I removed a lot of unnecessary `omitempty` in https://github.com/DataDog/dd-trace-go/pull/2869/commits/7c605497b6bc34274e08feabe9e21f98c2b4ef30